va/validation-authority.go

```diff
@@ -275,6 +279,14 @@ func (va *ValidationAuthorityImpl) fetchHTTP(ctx context.Context, identifier cor
 			Detail: fmt.Sprintf("Error reading HTTP response body: %v", err),
 		}
 	}
+	// io.LimitedReader will silently truncate a Reader so if the
+	// resulting payload is the same size as maxResponseSize fail
+	if len(body) >= maxResponseSize {
+		return nil, validationRecords, &probs.ProblemDetails{
+			Type:   probs.UnauthorizedProblem,
+			Detail: fmt.Sprintf("Invalid response from %s: response too large, \"%s\"", url.String(), body),
```

Comment:
Right. However, in 99.9% of cases where the response is >= 128 bytes, it's because the response doesn't match: it's actually HTML, or it's got some extra garbage, or whatever. Because we will now be catching such errors early with this length check, it's important to provide the most meaningful error.

The only case in which "response too large" would be the most meaningful error would be if they padded their response with lots of whitespace, which we expect to be very uncommon. However, even in that very uncommon case, the generic "Invalid response" error is accurate.